build/ci/get_image_tags_from_branch.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash -xe
+GIT_BRANCH=$1
+IMAGE_VERSION=$2
+BUILD_NUMBER=$3
+COMMIT_HASH=$4
+branch_tag=`echo $GIT_BRANCH| sed 's|/|.|g'`  #not sure if docker accept / in the version
+specific_tag="${IMAGE_VERSION}_b${BUILD_NUMBER}_${COMMIT_HASH}${branch_tag}"
```

Comment:
why the commit hash? and why is it not separated from the "branch_tag"?

Reply:
@ArbelNathan asked for it, and it can be cool to have the option to get a docker image by hash

Reply:
we pass the commit hash with `_` [here](https://github.com/IBM/ibm-block-csi-operator/blob/d51d2acf7702b751a0465fedb00b9f7a18c29d9a/build/ci/github_actions/setup_ci_dependencies.sh#L20)

Reply:
why? is the underscore part of the hash?

Reply:
and regrading the coolness, do we need the entire hash?
and how is it different than the jenkins build tag?

Reply:
if we don't have the commit info so I cant see why to save the images and not to delete them at the end of the tests. (just like CI-built images have the build number that point to the build that has the commit/branch info )

Reply:
> why? is the underscore part of the hash?

because in the Jenkins build, we don't pass the commit hash, and here we have the option for it and since jenkins builds are using the same [script](https://github.com/IBM/ibm-block-csi-operator/blob/task/CSI-2625_add_another_stage_before_the_merge/build/ci/get_image_tags_from_branch.sh) so I needed that it will work for both of them, so I added a `_` to the hash parameter

Reply:
> and regrading the coolness, do we need the entire hash?
>
no, because the first 7 digits are pretty uniq by themself
> and how is it different than the jenkins build tag?
>
because in Jenkins build tag we don't have the commit hash as part of the tag

Reply:
is it not possible to pass the commit hash in jenkins?
can't the script check if the hash is empty and only add `_` if it's not empty or something?

Reply:
> if we don't have the commit info so I cant see why to save the images and not to delete them at the end of the tests. (just like CI-built images have the build number that point to the build that has the commit/branch info )

I'm not sure it's relevant to this PR

Reply:
> is it not possible to pass the commit hash in jenkins?
>
I don't know, I didn't check, but we won't need it because you don't push every commit
> can't the script check if the hash is empty and only add `_` if it's not empty or something?
>
I can change it to this

Reply:
> I don't know, I didn't check, but we won't need it because you don't push every commit

I don't see how they are different, in jenkins, every commit is pushed for the main branches (e.g. develop), and can be pushed for every commit in the private branches.

> I can change it to this

that would be great, thanks

Reply:
> I don't see how they are different, in jenkins, every commit image is pushed for the main branches (e.g. develop), and can be pushed for every commit image in the private branches

I found a way to it and added it to the Jenkins pipeline

Reply:
> that would be great, thanks

done

Reply:
I still see `specific_tag="${IMAGE_VERSION}_b${BUILD_NUMBER}_${COMMIT_HASH}${branch_tag}"`

Reply:
that is because of [this](https://github.com/IBM/ibm-block-csi-operator/blob/773eada2d3347f333b33d682a9af30a518985003/build/ci/get_image_tags_from_branch.sh#L7)

Reply:
but if the jenkins could pass the commit hash as well, why should it be optional?

Reply:
removed it...

Reply:
now the first-7-chars slicing logic is duplicated (`:0:7`) - could you move it to `get_image_tags_from_branch`?

Reply:
done